### ISSUE TITLE
Wrong image link gets stored instead of becoming null when image is deleted

### ISSUE THREAD

**Issue by ShridharGoel** [fossasia/open-event-server#6303]
**Describe the bug**
On deleting profile image, the same image link gets stored in `avatarUrl` with `localhost` instead of `open-event-api-dev.herokuapp.com`.

**Expected behavior**
Image link should become null.

**Comment by shreyanshdwivedi** [comment]
Will take a look

**Comment by prateekj117** [comment]
@shreyanshdwivedi If you haven't spent much time on it, May I take this up?

**Comment by shreyanshdwivedi** [comment]
@prateekj117 yeah sure. I'm a bit busy with other issues

**Comment by prateekj117** [comment]
@shreyanshdwivedi @ShridharGoel I tried it in development (herokuapp) instance,  making `PATCH` request to change `avatar-url` from `non-null` to `null` works fine, and returns null.

**Comment by prateekj117** [comment]
@iamareebjamal

**Comment by mrsaicharan1** [comment]
@ShridharGoel this will be fixed here https://github.com/fossasia/open-event-frontend/pull/3361

**Comment by iamareebjamal** [comment]
But that is happening on mobile, so it is server issue. Fixing something in frontend won't fix it in mobile

**Comment by mrsaicharan1** [comment]
Okay.

**Comment by CosmicCoder96** [comment]
> the same image link gets stored in avatarUrl with localhost instead of open-event-api-dev.herokuapp.com.

@ShridharGoel  I don't follow? What's the correlation between image link has to do with localhost and heroku?

**Comment by iamareebjamal** [comment]
Image link is - open-event-api-dev.herokuapp.com/static/haha.jpeg

Becomes localhost/static/haha.jpeg on deleting it

**Comment by prateekj117** [comment]
> @shreyanshdwivedi @ShridharGoel I tried it in development (herokuapp) instance, making `PATCH` request to change `avatar-url` from `non-null` to `null` works fine, and returns `null`.

@iamareebjamal

**Comment by iamareebjamal** [comment]
@ShridharGoel Can you please test again

**Comment by ShridharGoel** [comment]
Tested again. The issue is still there.

**Comment by prateekj117** [comment]
@ShridharGoel If possible, can you please share the exact request being sent to the server.

**Comment by CosmicCoder96** [comment]
@iamareebjamal @ShridharGoel @prateekj117 I can't reproduce it through postman, does it happen only via android? It's strange because, deleting an image is equivalent to changing, the avatar URL, from some value to null, the keyword `localhost` can't magically appear.

@ShridharGoel How exactly are you testing it? If it's not reproducible via postman, then it's not a server issue.

**Comment by ShridharGoel** [comment]
@CosmicCoder96 After deleting an image, I am setting the avatar-url to null and updating the user. But then on fetching the user details again, the link with localhost is being fetched.

**Comment by iamareebjamal** [comment]
Try with Postman. If it is not happening with Postman, something is wrong on Android side

**Comment by ShridharGoel** [comment]
![Postman screenshot](https://user-images.githubusercontent.com/35566748/63194673-3e9ee300-c08e-11e9-9d7d-3c7fceaba50f.png)

**Comment by prateekj117** [comment]
@iamareebjamal I am not experiencing it using Postman.

**Comment by iamareebjamal** [comment]
Try with same payload @prateekj117

**Comment by iamareebjamal** [comment]
@ShridharGoel Does the same happen in production?

**Comment by prateekj117** [comment]
@iamareebjamal Tried. Still not happening locally. Though I found something weird. When I am sending this request: 

![Screenshot from 2019-08-17 13-26-50](https://user-images.githubusercontent.com/31065687/63208520-c4547a00-c0f2-11e9-944f-d39820d87ce0.png)

Only `avatar_url` gets updated, and the data in `before_update_object` function only has `avatar_url` in it. Can you please check if something is wrong in the request I am sending, or is it a bug.

**Comment by ShridharGoel** [comment]
> Does the same happen in production?

In production, the image is not getting deleted. Same link is being returned. Tried deleting the image via frontend too, but it did not get deleted.

**Comment by ShridharGoel** [comment]
> In production, the image is not getting deleted. Same link is being returned. Tried deleting the image via frontend too, but it did not get deleted.

@iamareebjamal Should I create a separate issue for this?

**Comment by prateekj117** [comment]
@ShridharGoel Try again after updating frontend dev branch, it was solved in https://github.com/fossasia/open-event-frontend/pull/3361.

> In production, the image is not getting deleted. Same link is being returned. Tried deleting the image via frontend too, but it did not get deleted.
@ShridharGoel Try again after updating frontend dev branch, it was solved in https://github.com/fossasia/open-event-frontend/pull/3361.

**Comment by ShridharGoel** [comment]
> Try again after updating frontend dev branch, it was solved in [fossasia/open-event-frontend#3361](https://github.com/fossasia/open-event-frontend/pull/3361).

But it's a server issue because it's there on Android too and can be reproduced via Postman also.

**Comment by prateekj117** [comment]
@ShridharGoel Yes. That is the case. But the localhost url is wrong anyway, so it should not have displayed the avatar after deletion. So it was happening on frontend due to some different issue I guess.

**Comment by ShridharGoel** [comment]
@prateekj117 It is displaying the same avatar after deletion only on the production instance. For the dev instance the link is getting changed to the `localhost` one.

**Comment by iamareebjamal** [comment]
@prateekj117 Frontend comes nowhere in this issue

**Comment by prateekj117** [comment]
@iamareebjamal Yes, that's a different issue.

**Comment by iamareebjamal** [comment]
All images except avatar-url and maybe thumbnail-url are/should be dump only. So, they should not be sent to server, 

Nevertheless, sending them should not break the implementation

**Comment by iamareebjamal** [comment]
@CosmicCoder96 Try uploading the image on https://open-event-fe.netlify.com

All links are localhost

**Comment by iamareebjamal** [comment]
Deleting is not working as well

**Comment by iamareebjamal** [comment]
I reproduced it the first time I tried. I don't know what you guys were doing. @prateekj117

**Comment by prateekj117** [comment]
@iamareebjamal I was trying on localhost.

**Comment by prateekj117** [comment]
@iamareebjamal In case you were able to reproduce it on localhost, can you please share your exact request.

**Comment by iamareebjamal** [comment]
Issue is on hosted instances
